nextcloud-startup-script.sh

```diff
@@ -330,27 +330,58 @@ bash $SCRIPTS/nextcloud_configuration.sh
 bash $SCRIPTS/additional_apps.sh
 
 clear
-# Change passwords
+### Change passwords
 # CLI USER
-print_text_in_color "$ICyan" "For better security, change the system user password for [$(getent group sudo | cut -d: -f4 | cut -d, -f1)]"
-any_key "Press any key to change password for system user..."
+msg_box "For better security, we will now change the password for the UNIX-user for Ubuntu"
+UNIXUSER="$(getent group sudo | cut -d: -f4 | cut -d, -f1)"
 while :
 do
-    sudo passwd "$(getent group sudo | cut -d: -f4 | cut -d, -f1)" && break
+    UNIX_PASSWORD=$(input_box_flow "Please type in the new password for the UNIX-user: [$UNIXUSER]")
+    if [[ "$UNIX_PASSWORD" == *" "* ]]
+    then
+        msg_box "Please don't use spaces"
+    else
+        break
+    fi
 done
-echo
-clear
+check_command echo "$UNIXUSER:$UNIX_PASSWORD" | sudo chpasswd
+unset UNIX_PASSWORD
+
 # NEXTCLOUD USER
 NCADMIN=$(occ_command user:list | awk '{print $3}')
-print_text_in_color "$ICyan" "The current admin user in Nextcloud GUI is [$NCADMIN]"
-print_text_in_color "$ICyan" "We will now replace this user with your own."
-any_key "Press any key to replace the current (local) admin user for Nextcloud..."
-# Create new user
+msg_box "We will now change the username and password for the Nextcloud admin user"
 while :
 do
-    print_text_in_color "$ICyan" "Please enter the username for your new user:"
-    read -r NEWUSER
-    sudo -u www-data $NCPATH/occ user:add "$NEWUSER" -g admin && break
+    NEWUSER=$(input_box_flow "Please type in the name of the Nextcloud admin user. It must differ from [$NCADMIN].\nThe only allowed character are: 'a-z', 'A-Z', '0-9', and '_.@-'")
+    if [[ "$NEWUSER" == *" "* ]]
+    then
+        msg_box "Please don't use spaces."
```

Comment:
We don't have to check spaces explicitely if we check for the pattern

Reply:
I'm not 100% sure about the pattern. What you say is correct, but I haven't tested the pattern enough to be 100% sure.

Busy with the official VM today, need to produce a POC.

Reply:
I thought the official was already working?

Reply:
Yes, but they want changes...

Reply:
7 hours straight now. Finally a first POC. 

Now I will eat and all that other important stuff. :D